Feature/upgrade packages all at once

sehqlr opened the pull request:
This fixes #1084 by adding a different playbook, with a different name, inspired by @BrianHicks. 

One thing that could make this PR better is to give the playbook a better name! Or perhaps other distinctions between the two use cases for upgrading system packages?

zoza1982 commented:
Looks good! Love it!

ryane commented:
maybe we rename the original playbook to something like 'rolling-upgrade-packages` and this one can just be 'upgrade-packages' or 'fast-upgrade-packages' ? Idk

BrianHicks commented:
I think the first solution hits the nail on the head actually, @ryane.

From | To
-----|---
upgrade-packages | rolling-upgrade-packages
upgrade-packages-all-at-once | upgrade-packages

sehqlr commented:
I renamed the packages as requested. The next step is to change the docs.

sehqlr commented:
Is the copy here good for the docs? I don't know how much detail we need to get into

ryane commented on docs/upgrading/index.rst:
```diff
@@ -11,6 +11,16 @@ However, upgrade support should be considered alpha at this time; it has not
+---------------------
+
+We provide two playbooks for upgrading OS-level system packages on a cluster
+after is has been provisioned: ``playbooks/upgrade-packages.yml`` and
```
"after **it** has been provisioned". But, maybe we just remove this phrase to avoid confusion? I am not sure how consistent we are but, in other places in the docs, we use the term "provision" to refer to the process of running Ansible on the cluster. Only the rolling upgrade would be safe on a "provisioned" cluster in that case.

ryane commented on docs/upgrading/index.rst:
```diff
@@ -11,6 +11,16 @@ However, upgrade support should be considered alpha at this time; it has not
+``playbooks/rolling-upgrade-packages.yml``. The first playbook upgrades each
+node on your cluster in parallel, and the second upgrades each node serially.
+You want the use the rolling upgrade on a cluster that is already running
+consul; otherwise, you will likely lose quorum and distablize your cluster.
```
destabilize

BrianHicks replied:
We should provide a sensible default here. Something like:

> When in doubt, prefer `playbooks/rolling-upgrade-packages.yml`.

ryane commented:
posted a couple of small comments; otherwise, lgtm

sehqlr commented:
Thanks for the look over. I need to turn on spell check, apparently.

sehqlr commented:
I made my edits: I removed the provision phrase (which wasn't needed), and I fixed not only my spelling error, but also a different one. I setup my editor to turn on the spell check for our docs filetypes, so hopefully I won't slip again.

BrianHicks commented:
LGTM
